lib/iris/tests/unit/coord_systems/test_RotatedPole.py

```diff
@@ -26,46 +23,42 @@ def setUp(self):
         )
 
     def test_crs_creation(self):
-        self.assertEqual(self.pole_lon, self.rp_crs.grid_north_pole_longitude)
-        self.assertEqual(self.pole_lat, self.rp_crs.grid_north_pole_latitude)
-        self.assertEqual(
-            self.rotation_about_new_pole, self.rp_crs.north_pole_grid_longitude
-        )
+        assert self.pole_lon == self.rp_crs.grid_north_pole_longitude
+        assert self.pole_lat == self.rp_crs.grid_north_pole_latitude
+        assert self.rotation_about_new_pole == self.rp_crs.north_pole_grid_longitude
 
     def test_as_cartopy_crs(self):
         if cartopy.__version__ < "0.12":
-            with mock.patch("warnings.warn") as warn:
-                accrs = self.rp_crs.as_cartopy_crs()
-                self.assertEqual(warn.call_count, 1)
+            with pytest.warns() as warn:
+                _ = self.rp_crs.as_cartopy_crs()
+                assert len(warn) == 1
```

Comment:
```suggestion
            with pytest.warns() as warn:
                _ = self.rp_crs.as_cartopy_crs()
            assert len(warn) == 1
```

https://docs.astral.sh/ruff/rules/pytest-warns-with-multiple-statements/

Reply:
https://github.com/SciTools/iris/pull/6627/commits/257b04aba73930fa4e9a26b0ca5ef5a601c8cf49